src/utils/restRequest/index.ts

```diff
@@ -31,24 +25,129 @@ const useGoLazyRequest: <R, C = unknown>(requestOptions: LazyRequestOptions<R, T
     context: C | undefined,
 } = useLazyRequest;
 
-type GoRequestOptions<RESPONSE, ERROR, ADDITIONAL_OPTIONS> = Omit<RequestOptions<RESPONSE, ERROR, ADDITIONAL_OPTIONS>, 'url'> & {
-    pathVariables?: Record<string, number | string | undefined>;
-} & ({
-    apiType?: 'go';
-    url: keyof goApiPaths;
-} | {
-    apiType: 'risk';
-    url: keyof riskApiPaths;
-})
-
-const useGoRequest: <RESPONSE>(
-    requestOptions: GoRequestOptions<RESPONSE, TransformedError, AdditionalOptions>
-) => {
-    response: RESPONSE | undefined;
+type GoApiResponseWithContent = {
+    'responses': {
+        '200': {
+            'content': {
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                'application/json': any;
+            }
+        }
+    }
+}
+
+type GoApiUrlQueryWithContent = {
+    'parameters': {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        'query': any;
+    }
+}
+
+type GoApiUrlPathVariablesWithContent = {
+    'parameters': {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        'path': any;
+    }
+}
+
+type RiskApiResponseWithContent = {
+    'responses': {
+        '200': {
+            'content': {
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                'application/json': any;
+            }
+        }
+    }
+}
+
+type RiskApiUrlQueryWithContent = {
+    'parameters': {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        'query': any;
+    }
+}
+
+type RiskApiUrlPathVariablesWithContent = {
+    'parameters': {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        'path': any;
+    }
+}
+
+type GoApiUrlPaths = keyof goApiPaths;
+type RiskApiUrlPaths = keyof riskApiPaths;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type GetGoApi<URL extends GoApiUrlPaths> = goApiPaths[URL] extends { get: any } ? goApiPaths[URL]['get'] : unknown;
```

Comment:
```suggestion
type GetGoApi<URL extends GoApiUrlPaths> = goApiPaths[URL] extends { get: any } ? goApiPaths[URL]['get'] : never;
```

Reply:
I f we do this, the response is resolved as `undefined` but we actually want it to be `unknown` so that it's easier to cast if we have missing typings